framework/src/main/java/org/checkerframework/common/basetype/BaseTypeChecker.java

```diff
@@ -262,25 +262,21 @@ public boolean shouldResolveReflection() {
             if (t instanceof InvocationTargetException) {
                 Throwable err = t.getCause();
                 String msg;
-                if (err instanceof CheckerError) {
-                    CheckerError ce = (CheckerError) err;
-                    if (ce.userError) {
-                        // Don't add another stack frame, just show the message.
-                        throw ce;
-                    } else {
-                        msg = err.getMessage();
-                    }
+                // TODO: Handle if UserError?
```

Comment:
Yes, we should handle `UserError` here to provide the user message without being wrapped in another exception.